File: 자료구조/10845.js
const readline = require('readline');
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});
let input = [];
let cnt = 0;
let N = 0;
let queue = [];

const isEmpty = () => {
  if (queue.length === 0) return true;
  else return false;
};

rl.on('line', (line) => {
  if (cnt === 0) N = parseInt(line.trim());
  else if (cnt <= N + 1) {
    input.push(line.trim().split(' '));
  }
  cnt++;
  if (cnt === N + 1) rl.close();
});
rl.on('close', () => {
  for (let i = 0; i < N; i++) {
    if (input[i][0] === 'push') queue.push(input[i][1]);
    else if (input[i][0] === 'pop') {
      if (isEmpty()) console.log(-1);
      else console.log(queue.shift());
    } else if (input[i][0] === 'size') console.log(queue.length);
    else if (input[i][0] === 'empty') {
      if (isEmpty()) console.log(1);
      else console.log(0);
    } else if (input[i][0] === 'front') {
      if (isEmpty()) console.log(-1);
      else console.log(queue[0]);
    } else if (input[i][0] === 'back') {
      if (isEmpty()) console.log(-1);
      else console.log(queue.at(-1));
    }
  }

  process.exit();
});

File: 자료구조withPy/10845.py
import sys
from collections import deque

N=int(input())
queue=deque([])
def isEmpty(tmp):
    if len(tmp)==0:return True
    else: return False
    
for _ in range(N):
    order=sys.stdin.readline().split()
    if order[0]=="push":queue.append(int(order[1]))
    if order[0]=="pop":              # q비어있으면 -1
        if(isEmpty(queue)):print(-1)
        else:print(queue.popleft())
    if order[0]=="size": print(len(queue))
    if order[0]=="empty":              # q비어있으면 1 아니면 0
        if(isEmpty(queue)):print(1)
        else:print(0)
    if order[0]=="front":            # q비어있으면 -1
        if(isEmpty(queue)):print(-1)
        else:print(queue[0])
    if order[0]=="back":             # q비어있으면 -1
        if(isEmpty(queue)):print(-1)
        else:print(queue[-1])